admin/resources/js/components/form/Checklist.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import get from "lodash/get";
+import { RegisterOptions, useFormContext } from "react-hook-form";
+import Fieldset from "../H2Components/Fieldset";
+import Checkbox from "./Checkbox";
+import InputWrapper from "../H2Components/InputWrapper";
+
+export type Checkbox = { value: string | number; label: string };
+
+export interface ChecklistProps {
+  /** Each input element will be given an id to match to its label, of the form `${idPrefix}-${value}` */
+  idPrefix: string;
+  /** Holds text for the legend associated with the checklist fieldset. */
+  legend: string;
+  /** The name of this form control.
+   * The form's value at this key should be of type Array<string|number>. */
+  name: string;
+  /** A list of value and label representing the checkboxes shown.
+   * The form will represent the data at `name` as an array containing the values of the checked boxes. */
+  items: Checkbox[];
+  /** Set of validation rules and error messages to impose on all input elements. */
+  rules?: RegisterOptions;
+  /** If a context string is provided, a small button will appear which, when toggled, shows the context string below the inputs. */
+  context?: string;
+  /** If true, all input elements in this fieldset will be disabled. */
+  disabled?: boolean;
+  /** If true, and this input is not required, 'Optional' will not be shown above the fieldset. */
+  hideOptional?: boolean;
+}
+
+/**
+ * Must be part of a form controlled by react-hook-form.
+ * The form will represent the data at `name` as an array, containing the values of the checked boxes.
+ */
+const Checklist: React.FunctionComponent<ChecklistProps> = ({
+  idPrefix,
+  legend,
+  name,
+  items,
+  rules = {},
+  context,
+  disabled,
+  hideOptional,
+}) => {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext();
+  // To grab errors in nested objects we need to use lodash's get helper.
+  const error = get(errors, name)?.message;
+  const required = !!rules.required;
+
+  return (
+    <Fieldset
+      legend={legend}
+      name={name}
+      required={required}
+      error={error}
+      context={context}
+      disabled={disabled}
+      hideOptional={hideOptional}
+    >
+      {items.map(({ value, label }) => {
+        const id = `${idPrefix}-${value}`;
+        return (
+          <InputWrapper
+            key={id}
+            inputId={id}
+            label={label}
+            // Don't show Required tag, error or context on individual input, as its handled by Fieldset.
+            required={false}
+            hideOptional
+          >
+            <input
+              style={{ order: -1 }}
```

Comment:
What's the purpose of the css property order required here?

Reply:
It puts the input (the checkbox) to the left of the label.